src/main/java/jpa/dao/Contract.java

```diff
@@ -0,0 +1,8 @@
+package jpa.dao;
+public interface Contract<T>
```

Comment:
Why is the name of the interface is  contract? Better FGenericRepository or GenericDao